File: tests/test_logger.py
from pathlib import Path

from akiles2d.logger import blank, log, title, write
from akiles2d.simrc import LoggerConfig


def test_logger_writes(tmp_path: Path):
  logfile = tmp_path / "log.txt"
  opts = LoggerConfig(logfile=str(logfile), screendebuglevel=10, filedebuglevel=0, linelength=80)
  title("Test Title", 1, opts)
  log("hello", "INF", 1, opts)
  write("details", 1, opts)
  blank(1, opts)
  assert logfile.exists()

File: matlab_port/tests_matlab/test_logger.m
function tests = test_logger
% Migration tests for logger utilities.
import akiles2d.logger.*
tests = functiontests(localfunctions);

function testLogWritesFile(testCase)
  tmp = fullfile(tempdir,'akiles2d_logger.txt');
  opts.logfile = tmp;
  opts.screendebuglevel = 10;
  opts.filedebuglevel = 0;
  opts.linelength = 80;
  akiles2d.logger.title('Test Title',1,opts);
  akiles2d.logger.log('hello','INF',1,opts);
  akiles2d.logger.write('details',1,opts);
  akiles2d.logger.blank(1,opts);
  verifyTrue(testCase,isfile(tmp));
end

end
